modules/administration-guide/partials/snip_preparing-images-for-a-restricted-environment.adoc

```diff
@@ -0,0 +1,82 @@
+
+* The `{prod-cli}` management tool for {prod-short} version {prod-ver} is available on your workstation. See xref:using-the-chectl-management-tool.adoc[].
+
+* The `curl`, `jq`, and `yq` tools are available on your workstation.
+
+* The `opm` tool is available on your workstation. See link:https://docs.openshift.com/container-platform/{ocp4-ver}/cli_reference/opm/cli-opm-install.html[OpenShift documentation - Installing the `opm` CLI].
+
+* Podman is available on your workstation. See link:https://podman.io/getting-started/installation[Podman documentation - Installing Podman].
+
+* Skopeo is available on your workstation. See: link:https://github.com/containers/skopeo/blob/main/install.md[Skopeo documentation - Installing Skopeo].
+ 
+
+
+
+.Procedure
+
+// NOTE: Additional commands to run to test on a `crc` instance:
+// ----
+// $ export my_registry='default-route-openshift-image-registry.apps-crc.testing'
+// $ skopeo login -u kubeadmin -p $(oc whoami -t) --tls-verify=false $my_registry
+// ----
+
+// FIXME: list all required images -- This is not required we can use mapping.txt
+// export packages="$(curl -fsSL {prod-olm-catalog-csv-yaml} | yq -r '.spec.relatedImages[] | .image' | tr [:space:] , | head -c -1)"
+
+
+include::example$snip_che-installing-che-in-a-restricted-environment-step-disable-signature-checks.adoc[]
+
+. Run the following commands on your workstation to set placeholders as environment variables:
++
+[subs="+attributes,+quotes"]
+----
+$ export my_registry="{my_internal_registry}"
+$ export my_catalog="airgap-{prod-id-short}"
+$ export my_tag="$my_registry/$my_catalog/{prod-olm-catalog-image}:v{ocp4-ver}"
+----
+
+. Run the following commands on your workstation from a location with unrestricted network access to mirror the image required for the {prod-short} operator to the local Podman registry:
++
+[subs="+attributes,+quotes"]
+----
+$ opm index prune \
+    --from-index "registry.access.redhat.com/redhat/{prod-olm-catalog-image}:v{ocp4-ver}" \
+    --packages "eclipse-che" \
+    --tag "$my_tag"
+$ skopeo inspect containers-storage:$my_tag
+----
+
+. Run the following commands on your workstation from a location with access to the destination registry to push the operator image to the destination registry and enable it:
++
+[subs="+attributes,+quotes"]
+----
+$ skopeo login $my_registry
+$ oc new-project $my_catalog
+$ skopeo copy --dest-tls-verify=false --all containers-storage:$my_tag docker://$my_tag
+$ oc adm catalog mirror --insecure --manifests-only $my_tag $my_registry
+$ oc apply -f manifests-community-operator-index-*/catalogSource.yaml
+----
+
+. Run the following commands on your workstation from a location with unrestricted network access to mirror the images required for {prod-short} to the local Podman registry:
++
+[subs="+attributes,+quotes"]
+----
+$ oc image mirror --insecure=true -f  manifests-community-operator-index-*/mapping.txt
```

Comment:
Shouldn't we move this step up? When you were running commands "from a location with unrestricted" network access.

Reply:
We needed to generate the mapping file first. New method avoids this step.